proto/pulumi/resource.proto

```diff
@@ -238,3 +239,11 @@ message TransformResponse {
     google.protobuf.Struct properties = 1; // the transformed input properties.
     TransformResourceOptions options = 2; // the options for the resource.
 }
+
+message RegisterDefaultProvidersRequest {
+    repeated string providers = 1; // the provider refs to register as the default providers.
+}
+
+message RegisterDefaultProvidersResponse {
+    string monitorPort = 1;
```

Comment:
I'd lean towards having this return the full grpc target, not just the port. Don't assume localhost in the SDKs.

Reply:
That matches what we do for things like loaders and mappers.

Reply:
Makes sense, will change that.